### ISSUE TITLE
Division by zero in CalculationTrait.php

### ISSUE THREAD

**Issue by avoelkl** [wirecard/magento2-ee#263]
Hi,

I've recently updated from version 1.5.1 to 1.5.3 due to an error in the Magento 2 checkout.

The error was:
``{"0":"Warning: Division by zero in \/magento\/releases\/17656\/vendor\/wirecard\/magento2-ee\/Controller\/Frontend\/Creditcard.php on line 267",....``

Now that I've updated to 1.5.3 it seems the error just moved to another file:

https://github.com/wirecard/magento2-ee/blob/0ef64083eb037f58fab064dd4f2f72992e0d2839/Gateway/Helper/CalculationTrait.php#L52

```{"0":"Warning: Division by zero in /magento\/releases\/17664\/vendor\/wirecard\/magento2-ee\/Gateway\/Helper\/CalculationTrait.php on line 52","1":"<pre>
#1 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard->divide() called at [vendor\/wirecard\/magento2-ee\/Gateway\/Helper\/CalculationTrait.php:68]\n
#2 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard->calculateTax() called at [vendor\/wirecard\/magento2-ee\/Controller\/Frontend\/Creditcard.php:374]\n
#3 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard->addOrderItemsToBasket() called at [vendor\/wirecard\/magento2-ee\/Controller\/Frontend\/Creditcard.php:332]\n
#4 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard->setAdditionalInformation() called at [vendor\/wirecard\/magento2-ee\/Controller\/Frontend\/Creditcard.php:298]\n
#5 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard->processCreditCard() called at [vendor\/wirecard\/magento2-ee\/Controller\/Frontend\/Creditcard.php:200]\n
#6 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard->execute() called at [vendor\/magento\/framework\/App\/Action\/Action.php:108]\n
#7 Magento\\Framework\\App\\Action\\Action->dispatch() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:58]\n
#8 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard\\Interceptor->___callParent() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:138]\n
#9 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard\\Interceptor->Magento\\Framework\\Interception\\{closure}() called at [vendor\/magento\/framework\/Interception\/Interceptor.php:153]\n
#10 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard\\Interceptor->___callPlugins() called at [generated\/code\/Wirecard\/ElasticEngine\/Controller\/Frontend\/Creditcard\/Interceptor.php:26]\n
#11 Wirecard\\ElasticEngine\\Controller\\Frontend\\Creditcard\\Interceptor->dispatch() called at [vendor\/magento\/framework\/App\/FrontController.php:159]\n
#12 Magento\\Framework\\App\\FrontController->processRequest() called at [vendor\/magento\/framework\/App\/FrontController.php:99]\n...
#22 Magento\\Framework\\App\\Http->launch() called at [vendor\/magento\/framework\/App\/Bootstrap.php:261]\n
#23 Magento\\Framework\\App\\Bootstrap->run() called at [pub\/index.php:40]\n<\/pre>","url":"\/at_de\/wirecard_elasticengine\/frontend\/creditcard","script_name":"\/index.php"}```

**Comment by serucee** [comment]
Hi,

we are sorry for the inconvenience. The issue is resolved in the 1.5.4 release.
Please update to the latest version.

Best regards,
Wirecard Shop Systems Team

**Comment by avoelkl** [comment]
Thanks a lot. I've updated to 1.5.4. and the issue is resolved.
